### PR TITLE
Fix issue with other Datadog instrumentations

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Add Datadog Classloader as TypePool for resolving type coming from DD instrumentations.

# Motivation
Interactions with AppSec or Tracer instrumentation leading to VerifyError in some circumstances

# Additional Notes

### CODE REVIEW

**Review by cimi — APPROVED** [review]
:+1: Thanks for diagnosing and fixing this!

I'm not familiar with ByteBuddy APIs but your explanation made sense and the code looks good to me.

File: dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/BaseIntegrationTest.java [hunk 1]
```diff
@@ -251,6 +253,27 @@ protected void assertCaptureThrowable(
     assertEquals(message, throwable.getMessage());
   }
 
+  protected static boolean logHasErrors(Path logFilePath, Function<String, Boolean> checker)
+      throws IOException {
+    final AtomicBoolean hasErrors = new AtomicBoolean();
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/BaseIntegrationTest.java:258]: to be able to mutate it inside the lambda below

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTransformer.java [hunk 2]
```diff
@@ -509,13 +508,22 @@ protected String getCommonSuperClass(String type1, String type2) {
       // duplicate class definition for name: "okhttp3/RealCall"
       // for more info see:
       // https://stackoverflow.com/questions/69563714/linkageerror-attempted-duplicate-class-definition-when-dynamically-instrument
-      ClassFileLocator locator =
-          AgentStrategies.locationStrategy().classFileLocator(classLoader, null);
-      TypePool tp =
+      TypePool tpCurrentClassLoader =
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTransformer.java]: done

File: dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/BaseIntegrationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cimi) [reply dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/BaseIntegrationTest.java:258]: Nit: we could `.reduce()` the stream to a boolean or `.filter()` and do a length check on the result to replace the `.forEach()` - IMO it's simpler and we remove the need for the extra var. It's fine to leave as is, up to you 🙂 

Also, is it intentional we don't print the failing line when the `checker` matches? We only print in the first if statement.

[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/src/test/java/datadog/smoketest/BaseIntegrationTest.java:258]: True, I will address this in a next PR.

**Pull request merged**